frontend/src/metabase/admin/people/containers/EditUserModal/EditUserModal.tsx

```diff
@@ -19,19 +23,28 @@ export const EditUserModal = ({ onClose, params }: EditUserModalProps) => {
   const dispatch = useDispatch();
   const userId = params.userId ? parseInt(params.userId) : null;
   const { data: user, isLoading } = useGetUserQuery(userId ?? skipToken);
-
+  const [updateUser] = useUpdateUserMutation();
   const initialValues = useMemo(() => getInitialValues(user), [user]);
 
-  const handleSubmit = async (newValues: Partial<UserType>) => {
-    // first name and last name keys need to be present so they can
-    // potentially be removed
-    const submitValues = {
+  const handleSubmit = async (newValues: Partial<User>) => {
+    if (userId == null) {
+      return;
+    }
+
+    // first name and last name keys need to be present, so they can potentially be removed
+    const newUser = await updateUser({
+      id: userId,
       first_name: null,
       last_name: null,
       ...newValues,
-    };
-    // can't use metabase/api hook here until the people list uses it too
-    await dispatch(Users.actions.update({ id: user?.id, ...submitValues }));
+    }).unwrap();
+
+    // for compatibility with code that relies on the entity framework
+    dispatch({
+      type: Users.actionTypes.UPDATE,
+      payload: { user: newUser },
+    });
```

Comment:
This feels a bit fragile to me. Why not put this in the RTK endpoint `onQueryStarted` itself so you don't have to remember to do this?

Reply:
Great idea, fixed. Could you take a look? cc @uladzimirdev

Reply:
I did something similar in session/properties

`frontend/src/metabase/api/session.ts`